lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
Yes, I reviewed pretty much everything that has been said so far. I probably don't fully grasp some of it.

All I was trying to say is that by using wp/v2/widgets/:type(:id_base whichever naming you prefer) for specifically handling create requests it actually matches up to the request.

$this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ) currently is blank for the master, so I am assuming we will be passing a create widget request like:

POST wp/v2/widgets/?type=pages&sidebar_id=sidebar-1&sidebar_position=1

Where if we left the endpoints the way they are in the master it can just be this:

POST wp/v2/widgets/pages/?sidebar_id=sidebar-1&sidebar_position=1

It doesn't matter to me really either way so if #17 is the way forward I am onboard and I will probably have Delete on a pull request tonight.  Let me know if I should be gearing that towards master or towards #17 .  Thanks!